apps/src/templates/UnitSelectorV2.jsx

```diff
@@ -0,0 +1,77 @@
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import React from 'react';
+import SimpleDropdown from '../componentLibrary/simpleDropdown/SimpleDropdown';
+import analyticsReporter from '@cdo/apps/lib/util/AnalyticsReporter';
+import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import {loadUnitProgress} from '@cdo/apps/templates/sectionProgress/sectionProgressLoader';
+import {setScriptId} from '@cdo/apps/redux/unitSelectionRedux';
+
+function UnitSelectorV2({
+  sectionId,
+  scriptId,
+  coursesWithProgress,
+  className,
+  setScriptId,
+}) {
+  const onSelectUnit = React.useCallback(
+    e => {
+      const newScriptId = e.target.value;
+      setScriptId(newScriptId);
+      loadUnitProgress(newScriptId, sectionId);
+
+      this.recordEvent('change_script_v2', {
+        old_script_id: scriptId,
+        new_script_id: newScriptId,
+      });
+
+      analyticsReporter.sendEvent(EVENTS.PROGRESS_V2_CHANGE_UNIT, {
+        sectionId: sectionId,
+        oldUnitId: scriptId,
+        unitId: newScriptId,
+      });
+    },
+    [scriptId, setScriptId, sectionId]
+  );
```

Comment:
The script/unit synonym makes this a bit hard to follow. Could you perhaps replace `script` with `unit` wherever it's easy? Like, `setScriptId` is probably used in a bunch of places so that can be a refactor for another time, but I'd love to see us standardize on `unit` for new local variables, event names, etc.

Reply:
the scriptId is used in 17 files and `setScriptId` is used in 16. I think I'd prefer to put that into a different PR. I did however change the newScriptId and also set `unitId = scriptId` in the file.